geckomat/utilities/FVA/comparativeFVA.m

```diff
@@ -109,52 +123,27 @@
 [~, stats] = plotCumDist(FVA_Dists,legends,titleStr);
 end
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-function [model,FluxDist] = constrainModel(model,gRate,posCS,chemostat,Cuptake)
-%Fix optimal carbon source uptake and growth rate values from the ecModel in the 
-% original model (Convention: GEMs represent uptake fluxes as negative values)
-[~,~,model] = constrainIrrevModel(model,true,gRate);
-if chemostat 
-    %Set glucose uptake rate as objective to minimize
-    model = setParam(model,'obj', posCS, 1);
-else
-    %Fix glucose uptake rate
-    model = setParam(model,'lb', posCS,-1.0001*Cuptake);
-    %model = setParam(model,'ub', posCS,-0.9999*Cuptake);
-end
-FluxDist = solveLP(model,1);
-FluxDist = FluxDist.x;
-if ~isempty(FluxDist)
-    Cuptake = FluxDist(posCS);
-    model   = setParam(model,'lb', posCS, 1.0001*Cuptake);
-    disp(['The optimal value for ' model.rxnNames{posCS} ' is: ' num2str(Cuptake)])
+function [OptimalValue, optFluxDist, model] = fixObjective(model,fixed,priorValue)
+% Optimize and fixes objective value for GEM
+objIndx  = find(model.c~=0);
+if nargin < 3
+    sol = solveLP(model);
+    priorValue = sol.x(objIndx);
 end
-end
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-function [OptimalValue, optFluxDist, irrevModel] = constrainIrrevModel(irrevModel,fixed,priorValue)
-% Optimize and fixes objective value for ecModel
-objIndx      = find(irrevModel.c~=0);
-optFluxDist  = [];
-OptimalValue = 0;
+
 if fixed 
     factor = 0.9999;
 else
     factor = 0;
 end
 
-if nargin == 3
-    irrevModel.lb(objIndx) = factor*priorValue;
-    irrevModel.ub(objIndx) = priorValue;
-else
-    sol = solveLP(irrevModel);
-    irrevModel.lb(objIndx) = factor*sol.x(objIndx);
-    irrevModel.ub(objIndx) = sol.x(objIndx);
-end
-sol = solveLP(irrevModel);
+model.lb(objIndx) = factor*priorValue;
+model.ub(objIndx) = 1.0001*priorValue;
```

Comment:
shouldn't this line only apply when `fixed=True`? As in other cases it could just be:
```matlab
model.ub(objIndx) = priorValue;
```
Or is it needed in all cases for simulation stability?

Reply:
When `fixed = True` the lower bound is also set, otherwise just an UB is set

Reply:
What I meant here is that, as I see it, when `fixed = True` it should be:
```matlab
model.lb(objIndx) = 0.9999*priorValue;
model.ub(objIndx) = 1.0001*priorValue;
```
whereas when `fixed=False` it should be:
```matlab
model.lb(objIndx) = 0;
model.ub(objIndx) = priorValue;
```
However, currently the `1.0001*priorValue` applies regardless of the value that `fixed` takes. Is that decision intentional or an oversight?

Reply:
It was intentional @BenjaSanchez :)